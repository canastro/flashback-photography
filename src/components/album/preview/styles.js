import {scale, rhythm} from '../../../utils/typography';
import presets from '../../../utils/presets';

export default {
    root: {
        display: 'block',
        backgroundColor: 'lightgray',
        flex: '1 0 0%',
        marginRight: rhythm(1 / 8),
        width: '100%',
        maxWidth: 290.1,
        position: 'relative',
        [presets.Tablet]: {
            marginRight: rhythm(1)
        },
        ':last-child': {
            marginRight: 0
        },
        color: 'inherit',
        textTransform: 'uppercase',
        textDecoration: 'none'
    },
    wrapper: {
        flexDirection: 'column',
        flexShrink: 0,
        position: 'relative',
        overflow: 'hidden',
        outline: 0
    },
    image: {
        margin: 0,
        height: '100%',
        width: '100%',
        verticalAlign: 'baseline',
        position: 'absolute',
        top: 0,
        bottom: 0,
        left: 0,
        right: 0,
        outline: 'none'
    },
    overlay: {
        position: 'absolute',
        top: 0,
        bottom: 0,
        left: 0,
        right: 0,
        backgroundColor: 'rgba(0,0,0,0.3)',
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
        color: 'white',
        ...scale(2 / 5)
    },
    icon: {
        fontSize: '90%',
        marginRight: rhythm(1 / 4)
    },
    title: {
        position: 'absolute',
        bottom: 0,
        height: 40,
        backgroundColor: 'white',
        width: '100%',
        opacity: 0.8,
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center'
    }
};
