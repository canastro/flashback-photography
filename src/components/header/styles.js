import {rhythm, scale} from '../../utils/typography';

export const burgerStyles = {
    bmBurgerButton: {
        position: 'absolute',
        width: '36px',
        height: '30px',
        left: 'initial',
        top: 20,
        right: 36
    },
    bmBurgerBars: {
        background: '#373a47'
    },
    bmCrossButton: {
        height: '24px',
        width: '24px'
    },
    bmCross: {
        background: '#bdc3c7'
    },
    bmMenuWrap: {
        top: 0
    },
    bmMenu: {
        background: '#373a47',
        padding: '2.5em 1.5em 0',
        fontSize: '1.15em'
    },
    bmMorphShape: {
        fill: '#373a47'
    },
    bmItemList: {
        color: '#b8b7ad',
        padding: '0.8em',
        display: 'flex',
        flexDirection: 'column'
    },
    bmOverlay: {
        background: 'rgba(0, 0, 0, 0.3)',
        left: 0,
        top: 0
    }
};

export default {
    root: {
        background: 'white',
        borderBottom: '1px solid rgba(0,0,0,0.08)',
        position: 'fixed',
        top: 0,
        left: 0,
        right: 0,
        zIndex: 1
    },
    wrapper: {
        maxWidth: 960,
        margin: '0 auto',
        overflow: 'hidden',
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'space-between'
    },
    titleLink: {
        padding: rhythm(3 / 4),
        display: 'inline-block',
        float: 'left',
        textDecoration: 'none'
    },
    title: {
        ...scale(4 / 5),
        lineHeight: 1,
        margin: 0,
        overflow: 'hidden'
    },
    titleSpan: {
        paddingLeft: `calc(${rhythm(1)} - 1px)`,
        borderLeft: '1px solid rgba(0,0,0,0.3)',
        lineHeight: 1,
        marginLeft: rhythm(1)
    },
    icon: {
        top: -4,
        display: 'inline-block',
        position: 'relative'
    },
    link: {
        color: 'inherit',
        display: 'inline-block',
        float: 'right',
        lineHeight: '35px',
        textDecoration: 'none',
        marginLeft: 20,
        '&:hover': {
            color: '#000'
        }
    }
};
