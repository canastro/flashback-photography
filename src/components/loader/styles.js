export default {
    root: {
        position: 'fixed',
        height: '100%',
        width: '100%',
        top: 0,
        left: 0,
        backgroundColor: 'black',
        opacity: 0.5,
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'center',
        zIndex: 10
    },
    indicator: {
        display: 'inline-block',
        color: '#fff',
        width: 100
    },
    svgIndeterminate: {
        animation: 'circular-rotate 1.4s linear infinite'
    },
    circle: {
        stroke: 'currentColor',
        strokeLinecap: 'round',
        animation: 'circular-dash 1.4s ease-in-out infinite',
        // Some default value that looks fine waiting for the animation to kicks in.
        strokeDasharray: '80,200',
        strokeDashoffset: 0
    },
    '@keyframes circular-rotate': {
        '100%': {
            transform: 'rotate(360deg)'
        }
    },
    '@keyframes circular-dash': {
        '0%': {
            strokeDasharray: '1,200',
            strokeDashoffset: 0
        },
        '50%': {
            strokeDasharray: '100,200',
            strokeDashoffset: -15
        },
        '100%': {
            strokeDasharray: '100,200',
            strokeDashoffset: -120
        }
    }
};
