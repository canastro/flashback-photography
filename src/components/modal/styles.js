import {rhythm} from '../../utils/typography';

export default {
    wrapper: {
        display: 'flex',
        position: 'relative',
        height: '100vh'
    },
    caretsWrapper: {
        display: 'flex',
        maxWidth: rhythm(40.25),
        width: '100%',
        marginTop: 20,
        marginLeft: 'auto',
        marginRight: 'auto'
    },
    caretLeft: {
        cursor: 'pointer',
        fontSize: '50px',
        color: 'rgba(255,255,255,0.7)',
        userSelect: 'none',
        marginTop: 215
    },
    caretRight: {
        cursor: 'pointer',
        fontSize: '50px',
        color: 'rgba(255,255,255,0.7)',
        userSelect: 'none',
        marginTop: 215
    },
    close: {
        cursor: 'pointer',
        color: 'rgba(255,255,255,0.8)',
        fontSize: '30px',
        position: 'absolute',
        top: rhythm(1 / 4),
        right: rhythm(1 / 4)
    }
};
