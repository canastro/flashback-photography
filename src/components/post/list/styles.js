import {rhythm, scale} from '../../../utils/typography';
import presets from '../../../utils/presets';

export default {
    root: {
        display: 'flex',
        alignItems: 'stretch',
        flexShrink: 0,
        flexDirection: 'column'
    },
    postsChunk: {
        display: 'flex',
        alignItems: 'stretch',
        flexShrink: 0,
        flexDirection: 'row',
        marginBottom: rhythm(1 / 8),
        [presets.Tablet]: {
            marginBottom: rhythm(1)
        }
    },
    showMore: {
        ...scale(-0.5),
        border: '1px solid black',
        boxShadow: 0,
        background: 'none',
        color: 'black',
        cursor: 'pointer',
        margin: '0 auto',
        padding: rhythm(1 / 2),
        width: `calc(100vw - ${rhythm(1)})`,
        marginLeft: rhythm(0.5),
        marginRight: rhythm(0.5),
        marginBottom: rhythm(0.5),
        marginTop: rhythm(0.5),
        [presets.Tablet]: {
            margin: '0 auto',
            marginBottom: rhythm(1.5),
            marginTop: rhythm(1.5),
            height: rhythm(2),
            width: rhythm(5),
            textAlign: 'center'
        },
        '&:hover': {
            backgroundColor: '#ccc'
        }
    }
};
