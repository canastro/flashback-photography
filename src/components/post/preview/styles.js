import {scale, rhythm} from '../../../utils/typography';
import presets from '../../../utils/presets';

export default {
    root: {
        display: 'block',
        backgroundColor: 'black',
        flex: '1 0 0%',
        marginRight: rhythm(1 / 8),
        width: '100%',
        maxWidth: 290.1,
        position: 'relative',
        [presets.Tablet]: {
            marginRight: rhythm(1)
        },
        ':last-child': {
            marginRight: 0
        }
    },
    wrapper: {
        width: '100%',
        height: '100%',
        display: 'flex',
        flexDirection: 'column',
        justifyContent: 'center'
    },
    image: {
        maxHeight: 200
    },
    overlay: {
        position: 'absolute',
        top: 0,
        bottom: 0,
        left: 0,
        right: 0,
        backgroundColor: 'rgba(0,0,0,0.3)',
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
        color: 'white',
        ...scale(2 / 5)
    },
    icon: {
        fontSize: '90%',
        marginRight: rhythm(1 / 4)
    }
};
