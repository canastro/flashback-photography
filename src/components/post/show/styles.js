import presets from '../../../utils/presets';
import typography, {rhythm, scale} from '../../../utils/typography';

export default {
    root: {
        maxHeight: 500,
        backgroundColor: 'black',
        display: 'flex',
        alignItems: 'stretch',
        flexDirection: 'column',
        width: '100%',
        [presets.Tablet]: {
            flexDirection: 'row-reverse'
        }
    },
    detailsContainer: {
        ...scale(-2 / 5),
        backgroundColor: 'white',
        lineHeight: typography.options.baseLineHeight,
        height: '100%'
    },
    detailsContent: {
        display: 'flex',
        flexDirection: 'column',
        justifyContent: 'space-around',
        height: 'calc(100% - 80px)'
    },
    date: {
        color: 'rgba(0,0,0,0.4)',
        float: 'right'
    },
    tabletDetailsContainer: {
        backgroundColor: 'white',
        padding: rhythm(3 / 4),
        paddingBottom: 0,
        [presets.Tablet]: {
            width: rhythm(13),
            padding: rhythm(1)
        }
    },
    tabletDetailsWrapper: {
        display: 'none',
        [presets.Tablet]: {
            display: 'block',
            height: '100%'
        }
    },
    mobileDetailsWrapper: {
        background: 'white',
        padding: rhythm(3 / 4),
        display: 'block',
        [presets.Tablet]: {
            display: 'none'
        }
    },
    imageContainer: {
        backgroundColor: 'lightgray',
        flex: '1 0 0',
        margin: 'auto',
        width: '100%'
    },
    listContainer: {
        display: 'flex',
        flexDirection: 'column'
    },
    settingItem: {
        listStyle: 'none',
        padding: 0,
        height: 25,
        display: 'flex',
        alignItems: 'center'
    },
    icon: {
        marginRight: 10
    },
    tag: {
        backgroundColor: '#ccc',
        margin: '10px 0 0 0',
        textAlign: 'center',
        width: 100,
        color: 'inherit',
        textDecoration: 'none',
        '&:hover': {
            backgroundColor: '#696969'
        }
    }
};
