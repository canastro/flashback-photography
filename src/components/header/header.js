// @flow
import React from 'react';
import CameraRetroIcon from 'react-icons/lib/fa/camera-retro';
import Link from 'gatsby-link';
import injectSheet from 'react-jss';
import {slide as Menu} from 'react-burger-menu';

import styles, {burgerStyles} from './styles';

type Props = {
  classes: Object
};

/**
 * Header component
 */
export class Header extends React.Component {
  props: Props;

  /**
   * Initialize state with isOpen false
   * @param {Object} props - react props
   */
  constructor(props) {
      super(props);

      this.state = {
          isOpen: false
      };
  }

  /**
   * Updates the state with the value sent from the hamburguer menu
   * @method handleMenuChange
   * @param {Object} state - menu state
   */
  handleMenuChange = (state: Object) => {
      this.setState(state);
  };

  /**
   * Sets the isOpen as false when a link is clicked
   * @method handleLinkClick
   */
  handleLinkClick = () => {
      this.setState({isOpen: false});
  };

  /**
   * Builds Links
   * @returns {Array<Node>} array
   */
  getLinks() {
      const {classes} = this.props;

      return [
          <Link
              key="contact"
              onClick={this.handleLinkClick}
              to="/contact/"
              className={classes.link}
          >
        Contact
          </Link>,

          <Link
              key="about"
              onClick={this.handleLinkClick}
              to="/about/"
              className={classes.link}
          >
        About
          </Link>,
          <Link
              key="album"
              onClick={this.handleLinkClick}
              to="/albums/"
              className={classes.link}
          >
        Albums
          </Link>
      ];
  }

  /**
   * Returns header component
   * @method render
   * @returns {Node} React Node
   */
  render() {
      const {classes} = this.props;
      const links = this.getLinks();
      return (
          <div className={classes.root}>
              <div className={classes.wrapper}>
                  <Link to="/" className={classes.titleLink}>
                      <h1 className={classes.title}>
                          <CameraRetroIcon className={classes.icon} />
                          <span className={classes.titleSpan}>Flashback</span>
                      </h1>
                  </Link>

                  <div className={classes.desktopMenu}>{links}</div>

                  <div className={classes.mobileMenu}>
                      <Menu
                          className={classes.mobileMenu}
                          right
                          pageWrapId="page-wrap"
                          outerContainerId="outer-container"
                          styles={burgerStyles}
                          isOpen={this.state.isOpen}
                          onStateChange={this.handleMenuChange}
                      >
                          {[
                              <Link
                                  key="home"
                                  to="/"
                                  onClick={this.handleLinkClick}
                                  className={classes.link}
                              >
                  Home
                              </Link>,
                              ...links
                          ]}
                      </Menu>
                  </div>
              </div>
          </div>
      );
  }
}

export default injectSheet(styles)(Header);
