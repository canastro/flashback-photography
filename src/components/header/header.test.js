import React from 'react';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';

import {Header} from './header';
import styles from './styles';
import {getClasses} from '../../utils/tests';

const classes = getClasses(styles);

it('should render correctly', () => {
    const test = shallow(<Header classes={classes} />);
    expect(shallowToJson(test)).toMatchSnapshot();
});

describe('on menu state change', () => {
    it('should update state', () => {
        const output = shallow(<Header classes={classes} />);
        const menu = output.find('Menu');

        expect(output.state().isOpen).toEqual(false);
        menu.prop('onStateChange')({isOpen: true});
        expect(output.state().isOpen).toEqual(true);
    });
});

describe('on menu link click', () => {
    it('should update state', () => {
        const output = shallow(<Header classes={classes} />);
        const link = output.find('GatsbyLink').at(2);

        output.setState({isOpen: true});

        expect(output.state().isOpen).toEqual(true);
        link.simulate('click');
        expect(output.state().isOpen).toEqual(false);
    });
});
