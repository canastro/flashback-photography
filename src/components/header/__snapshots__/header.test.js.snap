// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="root"
>
  <div
    className="wrapper"
  >
    <GatsbyLink
      className="titleLink"
      to="/"
    >
      <h1
        className="title"
      >
        <FaCameraRetro
          className="icon"
        />
        <span
          className="titleSpan"
        >
          Flashback
        </span>
      </h1>
    </GatsbyLink>
    <div
      className="desktopMenu"
    >
      <GatsbyLink
        className="link"
        key="contact"
        onClick={[Function]}
        to="/contact/"
      >
        Contact
      </GatsbyLink>
      <GatsbyLink
        className="link"
        key="about"
        onClick={[Function]}
        to="/about/"
      >
        About
      </GatsbyLink>
      <GatsbyLink
        className="link"
        key="album"
        onClick={[Function]}
        to="/albums/"
      >
        Albums
      </GatsbyLink>
    </div>
    <div
      className="mobileMenu"
    >
      <Menu
        bodyClassName=""
        burgerBarClassName=""
        burgerButtonClassName=""
        className="mobileMenu"
        crossButtonClassName=""
        crossClassName=""
        id=""
        isOpen={false}
        itemListClassName=""
        menuClassName=""
        morphShapeClassName=""
        noOverlay={false}
        onStateChange={[Function]}
        outerContainerId="outer-container"
        overlayClassName=""
        pageWrapId="page-wrap"
        right={true}
        styles={
          Object {
            "bmBurgerBars": Object {
              "background": "#373a47",
            },
            "bmBurgerButton": Object {
              "height": "30px",
              "left": "initial",
              "position": "absolute",
              "right": 36,
              "top": 20,
              "width": "36px",
            },
            "bmCross": Object {
              "background": "#bdc3c7",
            },
            "bmCrossButton": Object {
              "height": "24px",
              "width": "24px",
            },
            "bmItemList": Object {
              "color": "#b8b7ad",
              "display": "flex",
              "flexDirection": "column",
              "padding": "0.8em",
            },
            "bmMenu": Object {
              "background": "#373a47",
              "fontSize": "1.15em",
              "padding": "2.5em 1.5em 0",
            },
            "bmMenuWrap": Object {
              "top": 0,
            },
            "bmMorphShape": Object {
              "fill": "#373a47",
            },
            "bmOverlay": Object {
              "background": "rgba(0, 0, 0, 0.3)",
              "left": 0,
              "top": 0,
            },
          }
        }
        width={300}
      >
        <GatsbyLink
          className="link"
          key="home"
          onClick={[Function]}
          to="/"
        >
          Home
        </GatsbyLink>
        <GatsbyLink
          className="link"
          key="contact"
          onClick={[Function]}
          to="/contact/"
        >
          Contact
        </GatsbyLink>
        <GatsbyLink
          className="link"
          key="about"
          onClick={[Function]}
          to="/about/"
        >
          About
        </GatsbyLink>
        <GatsbyLink
          className="link"
          key="album"
          onClick={[Function]}
          to="/albums/"
        >
          Albums
        </GatsbyLink>
      </Menu>
    </div>
  </div>
</div>
`;
